### DISCUSSION TITLE
Unable to download from BBC iPlayer

### DISCUSSION THREAD

**Discussion by jezzastern** [aclap-dev/video-downloadhelper#520] [Help and Q&A]
### Your question

Below is a copy of the bug details when I try to downlad from the BBC iPlayer

{
  "version": "9.0.1.13",
  "target": "microsoft",
  "channel": "stable",
  "page_url": "undef",
  "error": {
    "details": "SideDownload error: : [in#0 @ 0000015231bc9a40] Error opening input: Server returned 403 Forbidden (access denied)\r\nError opening input file https://vod-dash-uk.live.cf.md.bbci.co.uk/usp/auth/vod/piff_abr_full_hd/8f36a3-m001yzrv/vf_m001yzrv_acc645f6-653e-485e-8fe8-4257bc121800.ism/pc_hd_abr_v2_dash_master.mpd?Expires=1716234804&Signature=gJ38PXdozGFb6xrViXq8CI4tC6MSGMl53vE2Y56XD-hQyOD1YudDEmhNU5GtVs5mZmNSCIahaVjtzodLHy6Ivq6yJ2eEfUju-oAVnsn3oDjIeAHme8m~CZYLp-P1Y3L57YtWPWQwlCextAxrBkb~sckwC0IVuCtm3IDwgtfQx~1Q5N~zVj0Gqp4SBGyxPtuiGI9j8yFwzinwNdm4XUvxzxClUDlVuOKC-MEBjrPaY-L8aWbGvm4BAnF0erYNtAr76fnDBI~ooJHp5UCBzpn3G8T~t8cMynYqF4pzAjN41K1MUt9GGZkOCfUF0G4p3AAKJDSh2AoqSvt06bnonCufbw__&Key-Pair-Id=K2VWLYKQ4HU1FJ.\r\nError opening input files: Server returned 403 Forbidden (access denied)\r\n",
    "downloadable_id": "downloadable_1001692201257654",
    "error": "coapp_failure",
    "id": "coapp_failure_ba8f2eba-4b67-4742-b703-a584dbca2e43"
  }
}

### Addon & CoApp version and details

version: 9.0.1.13
target: microsoft
channel: stable
lang: en-GB
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"64211c3502bb4dd0"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36 Edg/125.0.0.0
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "download_directory": "D:\\Shared Folder\\Video Downloadhelper",
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 3
}

**Comment by paulrouget** [comment]
Ah, another MPD error. I'll look into it, thank you!

**Reply by GrampaWildWilly** [reply]
I have on very infrequent occasions gone to the BBC web site & gotten Petroc Trelawney's commentary on the Vienna Philharmonic's New Year's Day concert.  I had no problems retrieving it with ffmpeg.  But that's from BBC3 radio.  Radio.  On the other hand, the BBC web site refuses to let you view anything from the Beeb TV.  It seems you need to have a TV license.  This is a concept that was utterly new to me until I ran up against it a couple of years ago.  This seems to be a rather expensive item, not to mention that you need to be a resident of The Isles to even get one.  I gather this is the price you pay for having no advertising in your TV programs.  We over here don't pay a TV license but the obnoxious advertising . . .  Whatever.  That's off topic.  Back on topic, I have never been able to watch anything from the BBC.  But listening has never posed a problem.  The BBC uses DASH for its radio content.  Like I said, I've always been able to get that with ffmpeg.  Assuming the BBC also uses DASH for its TV content, I suspect it would be possible for a duly licensed Brit to also get this video content with ffmpeg.  Too bad I can't guide you.

**Reply by GrampaWildWilly** [reply]
Here's a short tutorial using this archived radio broadcast as an example:

https://www.bbc.co.uk/sounds/play/m001z5dy

You'll note that this was archived on the site today, but they will remove this apparently on 2024/6/18.  So if you're reading this after that date, the link will no longer exist.  Too bad.  You missed out.

I posted an extensive tutorial on ffmpeg on our old Google forum.  If you're really interested, you might hunt that up.  This is a particularly terse, abridged version of that.

Here I'm collecting the URL for the DASH master manifest, the MPD file.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/464d1a9d-0f0f-46ed-b844-9d941181bdb5)

In order to properly use a master manifest, you have to run it through ffprobe:

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/15384089/ffprobe.txt)

Using that information, you can download the item using ffmpeg.  You'll note some rather unexpected errors in that ffprobe report.  They are, of course, repeated in the following ffmpeg log.  This is not surprising given that ffmpeg always begins by issuing its own ffprobe on the item.  I took a leap of faith here.  Despite the errors, ffprobe went ahead & generated a report on the item.  That encouraged me to try ffmpeg anyway.  The ffprobe report shows 2 different audio tracks.  I picked the one with the higher bit rate.  Curiously, the result doesn't have the bit rate reported by ffprobe.  A mystery, but not a particularly interesting one.

[BBC3 Program mp3 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/15384132/BBC3.Program.mp3.Log.txt)

Here's the file properties of the file I downloaded:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/93fc0135-347b-408b-a151-65099ce5930d)

And here it is playing in VLC.  Ha ha.  Kidding.  That would be rather boring, to show VLC playing an audio-only mp3.  I mostly sampled this, not listening to the whole thing.  Although, the Capriccio Espagnol is in this, one of my particular favorite pieces, so I'm listening to that.

Perhaps this can guide you to use ffmpeg to download a video from BBC TV.  Let us know how you get on.

**Reply by GrampaWildWilly** [reply]
I should add that the offerings from VDH on this radio show were quite obviously not useful for downloading it. So for the moment, you have to resort to ffmpeg.  However, the CoApp uses ffmpeg, & given my success downloading this content, I have every confidence Paul can fix up VDH to also work here.  Can he do it quickly?  I have no idea.  I don't work for VDH.  I'm just another user.  I don't know what Paul's priorities are with features & fixes to VDH/CoApp.  I also have no idea how complex any changes to VDH might be to add the necessary support.  In a word, don't hold your breath.  If you need whatever it is, you'll learn ffmpeg & get your content right now.  Otherwise, you must wait patiently until Paul tells us he's got it working.
